src/axom/primal/tests/primal_tetrahedron.cpp

```diff
@@ -623,6 +624,39 @@ TEST_F(TetrahedronTest, regularTetrahedron)
     EXPECT_EQ(primal::ON_POSITIVE_SIDE, primal::orientation(pt, tri));
   }
 }
+
+TEST_F(TetrahedronTest, checkSign)
+{
+  using QPoint = TetrahedronTest::QPoint;
+  using QTet = TetrahedronTest::QTet;
+
+  int indices[] = {0, 1, 2, 3};
+
+  for(int i = 0; i < this->numTetrahedra(); ++i)
+  {
+    QTet tet = this->getTet(i);
+    double expVolume = tet.signedVolume();
+
+    // Run sign check through all vertex permutations for the tetrahedron
+    do
+    {
+      QTet tetPermuted =
+        QTet(tet[indices[0]], tet[indices[1]], tet[indices[2]], tet[indices[3]]);
+
+      double preCheckVolume = tetPermuted.volume();
+
+      tetPermuted.checkAndFixOrientation();
+      EXPECT_NEAR(expVolume, tetPermuted.signedVolume(), this->EPS);
+
+      // Verify absolute value of volume is still the same
+      EXPECT_NEAR(axom::utilities::abs(preCheckVolume),
+                  tetPermuted.signedVolume(),
```

Comment:
Should we avoid calculating the signedVolume twice?

Reply:
> Should we avoid calculating the signedVolume twice?

This is part of a check to verify the absolute value of the volume is still the same _and_ now positive after calling `checkAndFixOrientation()`, so we want the signed volume after calling the function.

Reply:
The `abs()` is unnecessary in this case, a better name would be `preCheckAbsVolume`.

Reply:
I think we still calculate signed volume twice after calling `checkAndFixOrientation()`, once at line 649 and once here.

Reply:
> I think we still calculate signed volume twice after calling checkAndFixOrientation(), once at line 649 and once here.

Good catch!